File: nosh_now_apis/Models/OrderDetail.cs
namespace MyApp.Models
{
    public class OrderDetail : IModel
    {
        public int Id { get; set;}
        public int Quantity { get; set;}
        public double Price { get; set;}
        public int FoodId { get; set;}
        public int OrderId { get; set;}  
        public virtual Order Order{ get; set;}
        public virtual Food Food{ get; set;}
    }
}

File: nosh_now_application/lib/data/models/order_detail.dart
import 'package:nosh_now_application/data/models/food.dart';

class OrderDetail{
  int odId;
  int orderId;
  double price;
  int quantity;
  Food food;
  OrderDetail({
    required this.odId,
    required this.orderId,
    required this.price,
    required this.quantity,
    required this.food,
  });
  factory OrderDetail.fromJson(Map<dynamic, dynamic> json) {
    return OrderDetail(
        odId: json['id'],
        orderId: json['orderId'],
        price: json['price'] / 1.0,
        quantity: json['quantity'],
        food: Food.fromJson(json['food']));
  }

}
